### PR TITLE
Update report

### PR DESCRIPTION
**Pull request by Mobius1D**
PR Checklist

- [ ] Update NEWS.md?

### CODE REVIEW

File: docs/homepage/blog/index.md [hunk 1]
```diff
@@ -28,4 +28,6 @@
   - [Chapter10 Mountain Car.jl](/blog/notebooks_for_reinforcement_learning_an_introduction/Chapter10_Mountain_Car.jl)
   - [Chapter11 Counter Example.jl](/blog/notebooks_for_reinforcement_learning_an_introduction/Chapter11_Counter_Example.jl)
   - [Chapter12 Random Walk.jl](/blog/notebooks_for_reinforcement_learning_an_introduction/Chapter12_Random_Walk.jl)
-  - [Chapter13 Short Corridor.jl](/blog/notebooks_for_reinforcement_learning_an_introduction/Chapter13_Short_Corridor.jl)
\ No newline at end of file
+  - [Chapter13 Short Corridor.jl](/blog/notebooks_for_reinforcement_learning_an_introduction/Chapter13_Short_Corridor.jl)
+
+  - [Establish a General Pipeline for Offline Reinforcement Learning Evaluation (Summer OSPP Project 210370741) Mid-term Report](/blog/ospp_mid-term_report_210370741)
```
Comment (findmyway) [inline docs/homepage/blog/index.md:33]: Move this line to the top of this paragraph.

File: docs/homepage/blog/ospp_mid-term_report_210370741/index.md [hunk 2]
```diff
@@ -0,0 +1,348 @@
+@def title = "Establish a General Pipeline for Offline Reinforcement Learning Evaluation (Summer OSPP Project 210370741) Mid-term Report"
+@def description = """
+    This is a technical report of the summer OSPP project [Establish a General Pipeline for Offline Reinforcement Learning Evaluation](https://summer.iscas.ac.cn/#/org/prodetail/210370741?lang=en) used for mid-term evaluation. The report is split into the following parts:
+    - `Introduction`
+    - `Project Overview`
+    - `Implemented Datasets`
+    - `Implementation Details and Challenges Faced`
+    - `Implications`
+    - `Future Plan`
```
Comment (findmyway) [inline docs/homepage/blog/ospp_mid-term_report_210370741/index.md:9]: These lines will break the format. Just use a plain sentence here.

File: docs/homepage/blog/ospp_mid-term_report_210370741/index.md [hunk 3]
```diff
@@ -0,0 +1,348 @@
...
+@def is_enable_toc = true
+@def has_code = true
+@def has_math = true
+
+@def front_matter = """
+    {
+        "authors": [
+            "author":"Prasidh Srikumar",
+            "authorURL":"https://github.com/Mobius1D"
+            "affiliation":"",
```
Comment (findmyway) [inline docs/homepage/blog/ospp_mid-term_report_210370741/index.md:20]: Add your university here

File: docs/homepage/blog/ospp_mid-term_report_210370741/index.md [hunk 4]
```diff
@@ -0,0 +1,348 @@
...
+
+One of the most notorious challenges in RL is the lack of reliable environments for training RL agents. Offline RL has played a pivotal role in solving this problem by removing the need for the agent to interact with the environment to improve its policy over time. This brings forth the problem of having reliable tests to verify the performance of RL algorithms. Such tests are facilitated by standard datasets ([RL Unplugged](https://arxiv.org/abs/2006.13888), [D4RL](https://arxiv.org/abs/2004.07219) to name a few) that are used to train Offline RL agents and benchmark against other algorithms and implementations. [ReinforcementLearningDatasets.jl](https://github.com/JuliaReinforcementLearning/ReinforcementLearning.jl/tree/master/src/ReinforcementLearningDatasets) provides a simple solution to access various standard datasets that are available for Offline RL benchmarking across a variety of tasks.
```
Comment (findmyway) [inline docs/homepage/blog/ospp_mid-term_report_210370741/index.md:40]: You can use \dcite{...} to refer those papers in the bib file

File: docs/homepage/blog/ospp_mid-term_report_210370741/index.md [hunk 5]
```diff
@@ -0,0 +1,348 @@
...
+Make the following datasets available in RLDatasets.
+
+- [D4RL: Datasets for Deep Data-Driven Reinforcement Learning](https://github.com/rail-berkeley/d4rl)
+- [An Optimistic Perspective on Offline Reinforcement Learning (ICML, 2020)](https://github.com/google-research/batch_rl)
+- [d4rl-pybullet](https://github.com/takuseno/d4rl-pybullet)'
+- [RL Unplugged: Benchmarks for Offline Reinforcement Learning](https://github.com/deepmind/deepmind-research/tree/master/rl_unplugged)
+
+Make standard policies in [Benchmarks for Deep Off-Policy Evaluation](https://github.com/google-research/deep_ope) to be available in RLDatasets.jl.
+
+Implement an OPE method and select between a number of standard policies for a particular task using RLDatasets.jl.
```
Comment (findmyway) [inline docs/homepage/blog/ospp_mid-term_report_210370741/index.md:59]: What's `OPE` here?

File: docs/homepage/blog/ospp_mid-term_report_210370741/index.md [hunk 6]
```diff
@@ -0,0 +1,348 @@
...
+
+```julia
+using ReinforcementLearningDatasets
+ds = dataset(
+        "hopper-medium-replay-v0";
+        repo="d4rl")
+samples = take!(ds, 2)
+```
+
+The type (`D4RLDataSet`) returned by `dataset` is an iterable that returns batches of data based on the requirement that is specified.
```
Comment (findmyway) [inline docs/homepage/blog/ospp_mid-term_report_210370741/index.md:104]: iterable -> iterator

File: docs/homepage/blog/ospp_mid-term_report_210370741/index.md [hunk 7]
```diff
@@ -0,0 +1,348 @@
...
+
+Credits: https://github.com/takuseno/d4rl-pybullet
+
+```julia
+using ReinforcementLearningDatasets
+ds = dataset(
+        "hopper-bullet-mixed-v0";
+        repo="d4rl-pybullet",
+    )
+samples = take!(ds, 2)
```
Comment (findmyway) [inline docs/homepage/blog/ospp_mid-term_report_210370741/index.md:118]: Could you print the result of a batch from julia-repl? That would help others understand the result more easily.

File: docs/homepage/blog/ospp_mid-term_report_210370741/index.md [hunk 8]
```diff
@@ -0,0 +1,348 @@
...
+│  │  └─ register.jl
+│  ├─ d4rl_dataset.jl
+│  └─ d4rl_pybullet
+│     └─ register.jl
+├─ init.jl
+└─ rl_unplugged
+   ├─ atari
+   │  ├─ register.jl
+   │  └─ rl_unplugged_atari.jl
+   └─ util.jl
```
Comment (findmyway) [inline docs/homepage/blog/ospp_mid-term_report_210370741/index.md:204]: wrap this part in `` ``` ``

File: docs/homepage/blog/ospp_mid-term_report_210370741/index.md [hunk 9]
```diff
@@ -0,0 +1,348 @@
...
+    RLDatasets.d4rl_init()
+    RLDatasets.d4rl_pybullet_init()
+    RLDatasets.atari_init()
+    RLDatasets.rl_unplugged_atari_init()
+end
+```
+
+#### D4RL Datasets implementation
+
+The `register.jl` for d4rl dataset is located in `src/d4rl/d4rl` which registers the `DataDeps`. The following is an example code for the registration.
```
Comment (findmyway) [inline docs/homepage/blog/ospp_mid-term_report_210370741/index.md:219]: A basic introduction to `DataDeps` here would be great.

File: docs/homepage/blog/ospp_mid-term_report_210370741/index.md (hunk 7 above)
Comment (Mobius1D) [reply docs/homepage/blog/ospp_mid-term_report_210370741/index.md:118]: it is very long. But I could print out the types for information

File: docs/homepage/blog/ospp_mid-term_report_210370741/index.md [hunk 10]
```diff
@@ -0,0 +1,393 @@
...
+Credits: [D4RL](https://github.com/rail-berkeley/d4rl)
+
+```julia
+using ReinforcementLearningDatasets
+ds = dataset(
+        "hopper-medium-replay-v0";
+        repo="d4rl")
+```
+
+The type (`D4RLDataSet`) returned by `dataset` is an `Iterator` that returns batches of data based on the requirement that is specified.
```
Comment (findmyway) [inline docs/homepage/blog/ospp_mid-term_report_210370741/index.md:96]: Given that we've setup the docs of the `RLDatasets` package, better to provide a link to it in https://juliareinforcementlearning.org/docs/rldatasets/

File: docs/homepage/blog/ospp_mid-term_report_210370741/index.md [hunk 11]
```diff
@@ -0,0 +1,393 @@
...
+        [1, 2]
+    )
+samples = Iterators.take(ds, 2)
+```
+
+The output is similar to D4RL.
+
+### RL Unplugged Atari Dataset
+
+Added support for `RL Unplugged` atari datasets. The datasets that are stored in the form of `.tfrecord` are fetched into julia. Lazy loading with multi threading is implemented. This implementation is based on previous work in `TFRecord.jl`.
```
Comment (findmyway) [inline docs/homepage/blog/ospp_mid-term_report_210370741/index.md:151]: Add a link to TFRecord.jl ?

File: docs/homepage/blog/ospp_mid-term_report_210370741/index.md [hunk 12]
```diff
@@ -0,0 +1,393 @@
...
+        repo="d4rl-pybullet",
+    )
+samples = Iterators.take(ds, 2)
+
+```
+The output is similar to D4RL.
+
+### Google Research Atari DQN Replay Datasets
+
+Added support for `Google Research Atari DQN Replay Datasets`. Currently, the datasets are directly loaded into the RAM and therefore, it is advised to be used only with sufficient amount of RAM. Support for lazy parallel loading in a `Channel` will be given soon.
```
Comment (findmyway) [inline docs/homepage/blog/ospp_mid-term_report_210370741/index.md:133]: Could you roughly provide the least required memory?

File: docs/homepage/blog/ospp_mid-term_report_210370741/index.md [hunk 13]
```diff
@@ -0,0 +1,393 @@
...
+
+For supporting `RL Unplugged dataset` I had to learn about `.tfrecord` files, `Protocol Buffers`, `buffered Channels` and julia `multi threading` which was used in a lot of occasions. It took some time to grasp all the concepts but the final implementation, however, was based on already existing work in `TFRecord.jl`.
+
+All of this work wouldn't have been possible without the patient mentoring and vast knowledge of my mentor [Jun Tian](https://github.com/findmyway), who has been pivotal in the design and implementation of the package. His massive experience and beautifully written code has provided a lot of inspiration to the making of this package. His amicable nature and commitment to the users of the package by providing timely and detailed explanations to any issues or queries related to the package despite his time constraints, has provided a long standing example as a developer and as a person. I also thank all the developers of the packages that `RLDatasets.jl` depend upon.
```
Comment (findmyway) [inline docs/homepage/blog/ospp_mid-term_report_210370741/index.md:222]:
```suggestion
All of this work wouldn't have been possible without the patient mentoring and vast knowledge of my mentor [Jun Tian](https://github.com/findmyway), who has been pivotal in the design and implementation of the package. His massive experience and beautifully written code has provided a lot of inspiration to the making of this package. His amicable nature and commitment to the users of the package by providing timely and detailed explanations to any issues or queries related to the package despite his time constraints, has provided a long standing example as a developer and as a person. I also thank all the developers of the packages that `RLDatasets.jl` depends upon.
```

**Review by findmyway — CHANGES_REQUESTED** [review]

**Pull request merged**